policybot/dashboard/dashboard.go

```diff
@@ -88,73 +110,155 @@ func New(router *mux.Router, clientID string, clientSecret string) *Dashboard {
+		endEntry()
 
-	rt := &RegisteredTopic{
-		Title:       t.Title(),
-		Description: t.Description(),
-		URL:         basePath + t.URLSuffix(),
-		Subtopics:   d.handleSubtopics(t.Subtopics(), htmlRouter, apiRouter, basePath+t.URLSuffix()),
+	d.addEntry("Test Flakes", "Discover the wonderful world of test flakes.").
+		addPage("/flakes", flakes.Render).
+		endEntry()
+
+	d.addEntry("Features and Test Plans", "Get information on product features and associated test plans.").
+		addPage("/features", features.Render).
+		endEntry()
+
+	// home page
+	var homeEntries []home.Entry
+	for _, sbe := range d.entries {
+		homeEntries = append(homeEntries, home.Entry{
+			Title:       sbe.Title,
+			Description: sbe.Description,
+			URL:         sbe.URL,
+		})
 	}
+	home := home.New(homeEntries)
+	d.registerUIPage("/", home.Render)
+
+	// 'page not found' error page
+	router.NotFoundHandler = nf
+
+	// oauth support
+	router.HandleFunc("/login", d.oauthLogin)
+	router.HandleFunc("/githuboauthcallback", d.oauthCallback)
 
-	t.Configure(htmlRouter, apiRouter, newRenderContext(rt, d.primaryTemplates, d.errorTemplates), &Options{"istio"}) // TODO: eliminate istio default
+	// API endpoints
+	d.registerAPI("/api/maintainers/", maintainers.ListAPI)
 
-	return rt
+	return d
 }
 
-func (d *Dashboard) handleSubtopics(subtopics []Topic, htmlRouter *mux.Router, apiRouter *mux.Router, basePath string) []*RegisteredTopic {
-	var rt []*RegisteredTopic
-	for _, s := range subtopics {
-		rt = append(rt, d.registerTopic(s, htmlRouter, apiRouter, basePath))
+func (d *Dashboard) addEntry(title string, description string) *sidebarEntry {
+	newEntry := &sidebarEntry{
+		Title:       title,
+		Description: description,
+		Dashboard:   d,
 	}
-	return rt
+
+	d.currentEntry = newEntry
+	d.entries = append(d.entries, newEntry)
+
+	return newEntry
 }
 
-func (d *Dashboard) RegisterPageNotFound() {
-	d.htmlRouter.StrictSlash(true).
-		PathPrefix("/").
+func (d *Dashboard) registerUIPage(path string, render types.RenderFunc) *mux.Route {
+	return d.router.
+		StrictSlash(true).
 		Methods("GET").
+		Path(path).
 		HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			b := &bytes.Buffer{}
+			renderInfo, err := render(req)
+			if err != nil {
+				d.renderError(w, err)
+				return
+			}
+
+			entry := d.entryMap[mux.CurrentRoute(req)]
+
+			title := renderInfo.Title
+			if title == "" && entry != nil {
+				title = entry.Title
+			}
+
+			desc := ""
+			if entry != nil {
+				desc = entry.Description
+			}
 
 			info := templateInfo{
-				Title:       "Page Not Found",
-				Description: "Page Not Found",
+				Title:         title,
+				Description:   desc,
+				Content:       renderInfo.Content,
+				Control:       renderInfo.Control,
+				Entries:       d.entries,
+				SelectedEntry: entry,
 			}
 
-			if err := d.notFoundTemplates.Execute(b, info); err != nil {
-				util.RenderError(w, util.HTTPErrorf(http.StatusInternalServerError, "%v", err))
+			b := &bytes.Buffer{}
```

Comment:
Minor Nit you can declare this locally as `var b bytes.Buffer` and pass its reference. The struct itself (at least) would be stack allocated.